src/tests/java/htsjdk/samtools/SAMSequenceDictionaryTest.java

```diff
@@ -39,6 +41,55 @@
 import javax.xml.bind.Unmarshaller;
 
 public class SAMSequenceDictionaryTest {
+    
+    @Test void testIntervalParser() {
+        final int contigLength = 100;
+        final SAMSequenceRecord ssr1 = new SAMSequenceRecord("1", contigLength);
+        final SAMSequenceDictionary dict = new SAMSequenceDictionary(
+                Arrays.asList(ssr1));
+        Assert.assertEquals(dict.size(), 1);
+        Assert.assertEquals(
+                dict.parseInterval("1"),
+                new Interval("1", 1, contigLength)
+               );
+        Assert.assertEquals(
+                dict.parseInterval("1:10"),
+                new Interval("1", 10, 10)
+               );
+        Assert.assertEquals(
+                dict.parseInterval("1:10bp"),
+                new Interval("1", 10, 10)
+               );
+        Assert.assertEquals(
+                dict.parseInterval("1:10-10kb"),
+                new Interval("1", 10, contigLength)
```

Comment:
mb is not tested 
kb isn't tested to return the correct interval (only clipped)